pylint/checkers/typecheck.py

```diff
@@ -356,6 +356,12 @@ def _missing_member_hint(owner, attrname, distance_threshold, max_choices):
         "Emitted when a dict key is not hashable "
         "(i.e. doesn't define __hash__ method).",
     ),
+    "E1141": (
+        "Dict is not iterable",
```

Comment:
Let's add a bit more context here, since the dictionary is actually iterable, but it's going to raise an error if we don't call `items()` on it. So maybe something along the lines of `Unpacking a dictionary in iteration without calling .items()`?

Reply:
fixed